artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQSession.java

```diff
@@ -194,7 +195,14 @@ private boolean checkAutoCreateQueue(SimpleString queueName, boolean isTemporary
          try {
             if (!queueBinding.isExists()) {
                if (bindingQuery.isAutoCreateQueues()) {
-                  server.createQueue(queueName, RoutingType.ANYCAST, queueName, null, true, isTemporary);
+                  SimpleString queueNameToUse = queueName;
+                  SimpleString addressToUse = queueName;
+                  if (CompositeAddress.isFullyQualified(queueName.toString())) {
+                     CompositeAddress compositeAddress = CompositeAddress.getQueueName(queueName.toString());
+                     addressToUse = new SimpleString(compositeAddress.getAddress());
+                     queueNameToUse = new SimpleString(compositeAddress.getQueueName());
+                  }
```

Comment:
@gtully This looks good, except you also need to check that the address exists and if auto-create addresses is switched on.  You can get this information from the bindings query.

Reply:
server.createQueue() will create the address if it doesn't already exist.

Reply:
hmm, all the tests had a createProducer earlier that auto created the address.
With a createConsumer first I see the problem, the bindingQuery AddressInfo is null.
Passing a null routing type in that case and some tidy up in auto address creation in serverimpl works but the routingType comes from defaultConfiguration rather than the address settings in that case.
Question - should the caller figure the routingType from the addressSettings and if so should the logic for null routingType in createQueue get whacked?

Reply:
adding the logic to figure the routingType as the path of least resistance and the most explicit. The null routing type autocreateaddress case does not work in createQueue - maybe that is an indication that the relevant code can be removed.